codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/PaginatorResponseClassSpec.java

```diff
@@ -0,0 +1,528 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.codegen.poet.model;
+
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+import java.security.InvalidParameterException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.lang.model.element.Modifier;
+import software.amazon.awssdk.codegen.model.intermediate.IntermediateModel;
+import software.amazon.awssdk.codegen.model.intermediate.MemberModel;
+import software.amazon.awssdk.codegen.model.intermediate.OperationModel;
+import software.amazon.awssdk.codegen.model.intermediate.ShapeModel;
+import software.amazon.awssdk.codegen.model.service.PaginatorDefinition;
+import software.amazon.awssdk.codegen.poet.ClassSpec;
+import software.amazon.awssdk.codegen.poet.PoetExtensions;
+import software.amazon.awssdk.codegen.poet.PoetUtils;
+import software.amazon.awssdk.pagination.NextPageSupplier;
+import software.amazon.awssdk.pagination.Paginated;
+import software.amazon.awssdk.pagination.PaginatedItemsIterable;
+import software.amazon.awssdk.pagination.PaginatedResponsesIterator;
+import software.amazon.awssdk.pagination.SdkIterable;
+
+/**
+ * Java poet {@link ClassSpec} to generate the response class for sync paginated operations.
+ *
+ * Sample of a generated class with annotations:
+
+    public final class ListTablesPaginator implements Paginated&lt;ListTablesResponse, String> {
```

Comment:
Probably would remove this from the javadocs and just have it in the diff test.

Reply:
Ok, what is diff test? Do you mean a new test class to test the generation of sample paginators file?